PWGMM/Mult/Tasks/dndeta-hi.cxx

```diff
@@ -245,98 +284,81 @@ struct MultiplicityCounter {
           tracketas.push_back(track.eta());
         }
 
-        for (auto eta : tracketas) {
+        for (auto& eta : tracketas) {
           registry.fill(HIST("hrecdndeta"), Double_t(kDATA), Double_t(kMBAND), cent, z, eta);
         }
+
         for (auto& v0 : fullV0s) {
-          registry.fill(HIST("hV0Mass"), Double_t(kDATA), Double_t(kK0short), v0.mK0Short(), -1, -1);
-          registry.fill(HIST("hV0Mass"), Double_t(kDATA), Double_t(kLambda), -1, v0.mLambda(), -1);
-          registry.fill(HIST("hV0Mass"), Double_t(kDATA), Double_t(kAntilambda), -1, -1, v0.mAntiLambda());
+          registry.fill(HIST("hV0Count"), Double_t(kDATA), Double_t(kK0short), Double_t(kAll));
+          registry.fill(HIST("hV0Count"), Double_t(kDATA), Double_t(kLambda), Double_t(kAll));
+          registry.fill(HIST("hV0Count"), Double_t(kDATA), Double_t(kAntilambda), Double_t(kAll));
+
+          if (v0.v0radius() > v0radius &&
+              v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()) > v0cospa &&
+              TMath::Abs(v0.template posTrack_as<FiTracks>().eta()) < etadau &&
```

Comment:
Calling `.posTrack_as/.negTrack_as` several times creates temporary iterators on each invocation. Consider caching them and re-using: e.g. `auto ptrack = v0.template posTrack_as<FiTracks>();`. Additionally, you can simply use `std::abs()` instead of `TMath::Abs()`.